### ISSUE TITLE
Regression in styles evaluation in `1.4.x` and above

### ISSUE THREAD

**Issue by wereHamster** [callstack/linaria#597]
## Environment

- Linaria version: v2.0.0-alpha.2
- Bundler (+ version): rollup
- Node.js version: v10
- OS: macOS

## Description

In v1 code like this works. In v2 linaria fails with the error `classes is not defined.`:

```
const classes = {
    cell: css`
      opacity: 0;
    `,
  };

  const className = css`
  &:hover .${classes.cell}) {
    opacity: .2;
  }
  `;
```

## Reproducible Demo

https://github.com/wereHamster/linaria-v2-regression/blob/master/input.js

**Comment by jayu** [comment]
@wereHamster Which version of Linaria you used before the update to `2.0`? I observed that this syntax didn't work before, but I don't remember in which version.  

Btw. You should avoid using linaria in not top-level scope, if you move quotes code up, it will work.

**Comment by wereHamster** [comment]
v1.3.3.

What do you mean not in top-level scope? I've had a lot of code that is something like that:

```
function Component() {
  return <div className={css`<plain CSS here>`} />
}
```

Is that also not allowed? Do I need to pull all those css`` blocks out of the component into top-level scope?

(For reference, here is one such example: https://github.com/timvir/timvir/blob/ba2ee527466a5ab503ee61988171f8203c10a56c/src/components/ColorContrastInspector/internal/Cell.tsx)

**Comment by jayu** [comment]
@wereHamster Recently we've been thinking if we can restrict using linaria just to top-level scope. Some people try to use Linaria like this :`

```
function Component(props) {
  return <div className={css`color: ${props.color}`} />
}
```
Which will no work, because in built time we don't know what is the value of `props.colors`.
We've been thinking about restricting, but we didn't do it yet, because there might be many users that will have to refactor the codebase.  

According to current Linaria assumptions, this code works

```function Component() {
  return <div className={css`<plain CSS here>`} />
}
```
and this code should also work, but it doesn't (which is regression as you mentioned, not only in `2.0` but also in `1.4` probably)

```function Component() {
  const color = 'red'
  return <div className={css`color: ${color}`} />
}
```
So for now you don't have to pull everything into top-level scope, but we think that is generally a great idea to have styles defined even in separate file, or at least in top-level scope to make code better organized.

**Comment by jayu** [comment]
@wereHamster I adjusted the Issue name, I hope you don't mind :)

**Comment by jayu** [comment]
fixed via #604, released in `2.0.0-alpha.4` and `1.4.0-beta.9`
